libs/blocks/footer/footer.js

```diff
@@ -315,7 +315,12 @@ async function fetchFooter(url) {
 
 export default async function init(block) {
   const { prefix } = locale;
-  const url = getMetadata('footer-source') || `${prefix}/footer`;
+  let url = '';
+  if (contentRoot?.length) {
+    url = getMetadata('footer-source') || `${prefix}${contentRoot}/footer`;
+  } else {
+    url = getMetadata('footer-source') || `${prefix}/footer`;
+  }
```

Comment:
```suggestion
  const url = getMetadata('footer-source') || `${prefix}${contentRoot}/footer`;
```
Since the default value of `contentRoot` is empty string, I think this would work. WDYT?